torch/_functorch/_aot_autograd/collect_metadata_analysis.py

```diff
@@ -96,17 +99,23 @@ def inner(*flat_args):
         input_info: List[InputAliasInfo] = []
         output_info: List[OutputAliasInfo] = []
 
-        flat_f_args = pytree.tree_map(_to_fun, flat_args)
-
         prior_grad_enabled = torch.is_grad_enabled()
         prior_autocast_states = _get_autocast_states()
 
         # See Note [Disabling Functionalize TLS Above Python Functionalization]
         disable_above = torch._C._ExcludeDispatchKeyGuard(
             torch._C.DispatchKeySet(torch._C.DispatchKey.Functionalize)
         )
-        with disable_above, FunctionalTensorMode():
+
+        pre_dispatch_ctx = (
+            enable_pre_dispatch() if pre_dispatch else contextlib.nullcontext()
+        )
+
+        with disable_above, FunctionalTensorMode(
```

Comment:
hmm. I probably wouldn't bother changing this code, since we don't need to run pre-dispatch functionalization during the metadata / analysis pass. In theory they should give us the exact same metadata, but in case there are any bugs in pre-dispatch I would rather stick with one (only run aotdispatch functionalization)